contk/dataloader/multi_turn_dialog.py

```diff
@@ -288,3 +291,391 @@ def _load_data(self):
 					(key, invalid_num / vocab_num, oov_num / vocab_num, max(sent_length), \
 					cut_word_num / vocab_num, max(turn_length), cut_sent_num / sent_num))
 		return vocab_list, valid_vocab_len, data, data_size
+
+class SwitchboardCorpus(MultiTurnDialog):
+	'''A dataloder for Switchboard dataset.
+
+	        Arguments:
+	            file_path (str): a str indicates the dir of OpenSubtitles dataset.
+	            max_vocab_size (int): A cut-off threshold of ^UNK^ tokens. The most frequent
+	            	max_vocab_size tokens are kept, while others are replaced by ^<UNK>^.
+	            	Default: 10000.
+	            max_sen_length (int): All sentences longer than `max_sen_length` will be shortened
+	                to first `max_sen_length` tokens. Default: 50.
+	            session_window (int): The maximum number utterance kept in a session. Default: 10
+
+	        Refer to :class:`.MultiTurnDialog` for attributes and methods.
+
+	        Todo:
+	            * add references
+	        '''
+	def __init__(self, file_path, max_vocab_size=10000, max_sen_length=50, session_window=10):
```

Comment:
please use similiar api like `UbuntuCorpus`. Use `min_vocab_times` instead of `max_vocab_size`, `max_turn_length` instead of `session_window`

Reply:
Moreover, there should be an ``invalid_vocab_times``. This is a new feature of dataloader. Simply, **valid words** are for dataloader, **invalid words** are for metrics (but haven't been merge to master now.). You can see #63 for details. So you have to change your `_load_data` to be compatible with the new feature.